lightning-persister/src/lib.rs

```diff
@@ -0,0 +1,371 @@
+		std::io,
+	};
+	#[cfg(target_os = "windows")]
+	use {
+		lightning::get_event_msg,
+		lightning::ln::msgs::ChannelMessageHandler,
+	};
+
+	struct TestWriteable{}
+	impl DiskWriteable for TestWriteable{
+		fn write(&self, writer: &mut fs::File) -> Result<(), Error> {
+			writer.write_all(&[42; 1])
+		}
+	}
+
+	#[test]
+	fn test_filesystem_persister() {
+		// Create the nodes, giving them FilesystemPersisters for data persisters.
+		let persister_0 = FilesystemPersister::new("persister0".to_string());
+		let persister_1 = FilesystemPersister::new("persister1".to_string());
+		let chanmon_cfgs = create_chanmon_cfgs(2);
+		let mut node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+		let chain_mon_0 = test_utils::TestChainMonitor::new(Some(&chanmon_cfgs[0].chain_source), &chanmon_cfgs[0].tx_broadcaster, &chanmon_cfgs[0].logger, &chanmon_cfgs[0].fee_estimator, &persister_0);
+		let chain_mon_1 = test_utils::TestChainMonitor::new(Some(&chanmon_cfgs[1].chain_source), &chanmon_cfgs[1].tx_broadcaster, &chanmon_cfgs[1].logger, &chanmon_cfgs[1].fee_estimator, &persister_1);
+		node_cfgs[0].chain_monitor = chain_mon_0;
+		node_cfgs[1].chain_monitor = chain_mon_1;
+		let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+		let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+		// Check that the persisted channel data is empty before any channels are
+		// open.
+		let mut persisted_chan_data_0 = persister_0.load_channel_data::<EnforcingChannelKeys>().unwrap();
+		assert_eq!(persisted_chan_data_0.keys().len(), 0);
+		let mut persisted_chan_data_1 = persister_1.load_channel_data::<EnforcingChannelKeys>().unwrap();
+		assert_eq!(persisted_chan_data_1.keys().len(), 0);
+
+		// Helper to make sure the channel is on the expected update ID.
+		macro_rules! check_persisted_data {
+			($expected_update_id: expr) => {
+				persisted_chan_data_0 = persister_0.load_channel_data::<EnforcingChannelKeys>().unwrap();
+				assert_eq!(persisted_chan_data_0.keys().len(), 1);
+				for mon in persisted_chan_data_0.values() {
+					assert_eq!(mon.get_latest_update_id(), $expected_update_id);
+				}
+				persisted_chan_data_1 = persister_1.load_channel_data::<EnforcingChannelKeys>().unwrap();
+				assert_eq!(persisted_chan_data_1.keys().len(), 1);
+				for mon in persisted_chan_data_1.values() {
+					assert_eq!(mon.get_latest_update_id(), $expected_update_id);
+				}
+			}
+		}
+
+		// Create some initial channel and check that a channel was persisted.
+		let _ = create_announced_chan_between_nodes(&nodes, 0, 1, InitFeatures::known(), InitFeatures::known());
+		check_persisted_data!(0);
+
+		// Send a few payments and make sure the monitors are updated to the latest.
+		send_payment(&nodes[0], &vec!(&nodes[1])[..], 8000000, 8_000_000);
+		check_persisted_data!(5);
+		send_payment(&nodes[1], &vec!(&nodes[0])[..], 4000000, 4_000_000);
+		check_persisted_data!(10);
+
+		// Close the channel and make sure everything is persisted as expected.
```

Comment:
Is this a reference to line 249 or is a check needed after line 240, too?

Reply:
I rearranged the comments to hopefully be more clear